tests/grid/filtered_iterator_06_operator.cc

```diff
@@ -51,28 +52,47 @@ test()
       ++i;
     }
 
-  // Count the faces that are on the boundary and have a manifold_id of 0
+  // Count the faces that are on the boundary and have a boundary_id of 0 and a
+  // manifold_id of 0
+  const types::boundary_id boundary_id = 0;
   const types::manifold_id manifold_id = 0;
   std::set<typename Triangulation<dim, spacedim>::active_face_iterator>
-    face_set;
+    boundary_face_set, manifold_face_set;
+
   for (const auto &cell : tria.active_cell_iterators())
     for (const unsigned int face_n : GeometryInfo<dim>::face_indices())
-      if (cell->face(face_n)->at_boundary() &&
-          cell->face(face_n)->manifold_id() == manifold_id)
-        face_set.insert(cell->face(face_n));
+      if (cell->face(face_n)->at_boundary())
+        {
+          if (cell->face(face_n)->boundary_id() == boundary_id)
+            boundary_face_set.insert(cell->face(face_n));
+
+          if (cell->face(face_n)->manifold_id() == manifold_id)
+            manifold_face_set.insert(cell->face(face_n));
+        }
+
+  std::size_t n_boundary_filtered_cells = 0;
+  for (const auto &filtered_cell :
+       tria.active_face_iterators() | IteratorFilters::AtBoundary() |
+         IteratorFilters::BoundaryIdEqualTo(boundary_id))
```

Comment:
:-) That's just elegant!

Reply:
Yeah, I'm really happy that you implemented `operator|`. It's so nice to chain these filters!

Reply:
I really like that syntax. It's been used on the unix command line for 50 years, of course, and I like that we'll be able to use it in C++ as well (or at least will be with C++20).